src/vs/base/browser/ui/positronComponents/splitters/verticalSplitter.tsx

```diff
@@ -8,154 +8,425 @@ import 'vs/css!./verticalSplitter';
-	showResizeIndicator?: boolean;
-	onBeginResize: () => VerticalSplitterResizeParams;
-	onResize: (width: number) => void;
-}) => {
+export const VerticalSplitter = ({
+	configurationService,
+	invert,
+	showSash,
+	collapsible,
+	onCollapsedChanged,
+	onBeginResize,
+	onResize,
+}: VerticalSplitterProps) => {
+	// Reference hooks.
+	const sashRef = useRef<HTMLDivElement>(undefined!);
+	const expandCollapseButtonRef = useRef<HTMLButtonElement>(undefined!);
+
 	// State hooks.
-	const [resizing, setResizing] = useState(false);
+	const [splitterWidth, setSplitterWidth] = useState(
+		calculateSplitterWidth(configurationService, collapsible)
+	);
+	const [sashWidth, setSashWidth] = useState(
+		calculateSashWidth(configurationService, collapsible)
+	);
+	const [sashIndicatorWidth, setSashIndicatorWidth] = useState(getSashSize(configurationService));
+	const [hoverDelay, setHoverDelay] = useState(getHoverDelay(configurationService));
+	const [hovering, setHovering] = useState(false);
+	const [highlightExpandCollapse, setHighlightExpandCollapse] = useState(false);
+	const [hoveringDelayer, setHoveringDelayer] = useState<Delayer<void>>(undefined!);
+	const [collapsed, setCollapsed] = useState(false);
+	const [resizeParams, setResizeParams] = useState<ResizeParams | undefined>(undefined);
+
+	// Main useEffect.
+	useEffect(() => {
+		// Create the disposable store for cleanup.
+		const disposableStore = new DisposableStore();
+
+		// Add the onDidChangeConfiguration event handler.
+		disposableStore.add(
+			configurationService.onDidChangeConfiguration(configurationChangeEvent => {
+				// When workbench.sash.* changes, update state.
+				if (configurationChangeEvent.affectsConfiguration('workbench.sash')) {
+					// Track changes to workbench.sash.size.
+					if (configurationChangeEvent.affectedKeys.has('workbench.sash.size')) {
+						setSplitterWidth(calculateSplitterWidth(configurationService, collapsible));
+						setSashWidth(calculateSashWidth(configurationService, collapsible));
+						setSashIndicatorWidth(getSashSize(configurationService));
+					}
+
+					// Track changes to workbench.sash.hoverDelay.
+					if (configurationChangeEvent.affectedKeys.has('workbench.sash.hoverDelay')) {
+						setHoverDelay(getHoverDelay(configurationService));
+					}
+				}
+			})
+		);
+
+		// Set the hover delayer.
+		setHoveringDelayer(disposableStore.add(new Delayer<void>(0)));
+
+		// Return the cleanup function that will dispose of the disposables.
+		return () => disposableStore.dispose();
+	}, [collapsible, configurationService]);
+
+	/**
+	 * Sash onPointerEnter handler.
+	 * @param e A PointerEvent that describes a user interaction with the pointer.
+	 */
+	const sashPointerEnterHandler = (e: PointerEvent<HTMLDivElement>) => {
+		hoveringDelayer.trigger(() => {
+			setHovering(true);
+			const rect = sashRef.current.getBoundingClientRect();
+			if (e.clientY >= rect.top + 4 && e.clientY <= rect.top + 4 + 25) {
```

Comment:
Worth a comment on the source of the adjustment factors in the comparison, i.e. 4 and 25?